static/js/stream_create.js

```diff
@@ -307,99 +330,71 @@ export function show_new_stream_modal() {
         $("#announce-new-stream").hide();
     }
     clear_error_display();
-
-    $("#stream-checkboxes label.checkbox").on("change", function (e) {
-        const elem = $(this);
-        const stream_id = Number.parseInt(elem.attr("data-stream-id"), 10);
-        const checked = elem.find("input").prop("checked");
-        const subscriber_ids = new Set(peer_data.get_subscribers(stream_id));
-
-        $("#user-checkboxes label.checkbox").each(function () {
-            const user_elem = $(this);
-            const user_id = Number.parseInt(user_elem.attr("data-user-id"), 10);
-
-            if (subscriber_ids.has(user_id)) {
-                user_elem.find("input").prop("checked", checked);
-            }
-        });
-
-        e.preventDefault();
-    });
 }
 
 export function create_handlers_for_users(container) {
     // container should be $('#people_to_add')...see caller to verify
-    container.on("change", "#user-checkboxes input", update_announce_stream_state);
-
-    // 'Check all' and 'Uncheck all' visible users
-    container.on("click", ".subs_set_all_users", (e) => {
-        $("#user-checkboxes .checkbox").each((idx, li) => {
-            if (li.style.display !== "none") {
-                $(li.firstElementChild).prop("checked", true);
+    function update_checked_state_for_users(value, users) {
```

Comment:
Can you add a comment explaining this function's role?  Something like:
"Update the `all_users` backing data structure for which users will be submitted should the user click save, and also ensure that any visible checkboxes reflect the state of that data structure"